File: support_tools/swigwin-3.0.12/Examples/test-suite/lua/smart_pointer_rename_runme.lua
require("import")	-- the import fn
import("smart_pointer_rename")	-- import lib into global
spr=smart_pointer_rename --alias

-- catching undefined variables
local env = _ENV -- Lua 5.2
if not env then env = getfenv () end -- Lua 5.1
setmetatable(env, {__index=function (t,i) error("undefined global variable `"..i.."'",2) end})


foo = spr.Foo()
assert(foo:ftest1(1) == 1)
assert(foo:ftest2(1,2) == 2)

bar = spr.Bar(foo)
assert(bar:test() == 3)
assert(bar:ftest1(1) == 1)
assert(bar:ftest2(1,2) == 2)

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/smart_pointer_rename_runme.go
package main

import . "./smart_pointer_rename"

func main() {
	f := NewFoo()
	b := NewBar(f)

	if b.Test() != 3 {
		panic(0)
	}

	if b.Ftest1(1) != 1 {
		panic(0)
	}

	if b.Ftest2(2, 3) != 2 {
		panic(0)
	}
}
